### PR TITLE
chore: upgrade stripe v8.x

### LINKED ISSUE #80: Support the new Stripe configuration object
### Features:

The `stripe-node` library supports a configuration object now. We can rewrite a bit of our logic to support that configuration object _and_ support some values that we don't support currently (such as host and port.)

### PR DESCRIPTION
**Pull request by 9renpoto**
### All Submissions:

- [x] Have you followed the guidelines in our Contributing section?
- [x] Have you checked to ensure there aren't other open
      [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you linted your code locally prior to submission?

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like
      us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

---

*Add any extra details about your changes*

**Edit by @dhaspden:** Resolves #80

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,14 +30,13 @@
     "@nestjs/testing": "6.10.13",
     "@types/jest": "24.0.25",
     "@types/node": "13.1.4",
-    "@types/stripe": "7.13.19",
```
Comment (9renpoto) [inline package.json]: Added Type Definitions https://github.com/stripe/stripe-node/releases/tag/v8.0.0

Comment (dhaspden) [reply package.json]: Can you update this dependency in the example app as well?

File: src/interfaces/StripeOptions.ts [hunk 2]
```diff
@@ -1,10 +1,5 @@
-import * as ProxyAgent from 'https-proxy-agent';
+import Stripe from 'stripe';
```
Comment (dhaspden) [inline src/interfaces/StripeOptions.ts]: Can you do `import { StripeConfig } from 'stripe';` and update the below

Comment (9renpoto) [reply src/interfaces/StripeOptions.ts]: I think that is better, too :)

https://github.com/stripe/stripe-node/blob/9d817130dd22312f7d1905fafccca45610df0e46/types/lib.d.ts#L7-L11

`StripeConfig` depends on `namespace Stripe`, cant export only `StripeConfig` :'(

File: package.json (hunk 1 above)
Comment (9renpoto) [reply package.json]: thank you review, fix it https://github.com/dhaspden/nestjs-stripe/pull/82/commits/e4629d4e33f5b09c40b75fa9265fda763e253f2a :)

**Comment by dhaspden** [comment]
When I run this locally I get a bunch of deprecation warnings when I run `yarn test`, see below:

```
(node:17938) Stripe: `setAppInfo` is deprecated. Use the `appInfo` config option instead.
```

We _should_ probably fix that before we release the next version. It doesn't necessarily need to be in this pull request though. I'll create an issue for it.

File: src/interfaces/StripeOptions.ts (hunk 2 above)
Comment (dhaspden) [reply src/interfaces/StripeOptions.ts]: Interesting! Okay I guess I can live with that 😆

**Pull request merged**